### PR TITLE
Add bulk refile controls

### LINKED ISSUE #289: Batch process unfiled images
Assign the images a randomized non-PII identifier.  The identifier "needs to be alphanumeric and randomly generated" with a structure specified by a configuration parameter.

We may want to have an option to assign a selected list of images to a single case or have all images be separate cases (e.g., do all selected images end up in the same folder or in separate folders).

### PR DESCRIPTION
**Pull request by naglepuff**
Resolves #289 

Now images can be refiled from the `Unfiled` directory in bulk. 

Selecting any number of images in the `Unfiled` directory will cause a new set of controls to appear at the top of the page, above the image table (putting them below the image table would cause the new controls to render beyond the limit of the scroll-able div if there were many images in the folder).

bulk_refile_options

If one wishes to file images under new TokenIDs, there are two options: together and separate. Filing together under new token IDs will store all selected images in a new subdirectory in `AvailableToProcess` using a new, randomly generated token. Filing separately will create new subdirectories for each selected image.

Random token generation is driven by a new setting in the `girder.local.conf` file: `new_token_pattern`. As a first pass at making it easy to define a pattern from which to create new token IDs, this should be an alphanumeric string that also contains any number of the characters `#` and `@`. A `#` character will be replaced with a random digit, and a `@` character will be replaced with a random letter. For example, if the setting had the value `100@@####`, an example of a random token ID could be `100GI3825`.

If any images in the `Unfiled` directory have upload information stored (as `item['wsi_uploadInfo']`), an option to file selected images together under an existing token becomes available.

bulk_refile_existing_token

### CODE REVIEW

File: docs/CUSTOMIZING.rst [hunk 1]
```diff
@@ -127,6 +127,15 @@ The target text column should contain label text of WSIs in the import directory
...
 If ``InputFileName`` is added to the list of export fields in the ``upload_metadata_for_export_report`` settings, then the original file name will be included in the export report.
 
+Creating New TokenIDs for Refiling Images
++++++++++++++++++++++++++++++++++++++++++
+
+When using the bulk refile controls to move images from the ``Unfiled`` directory to ``AvailableToProcess``, the system can automatically generates new TokenIDs. The new TokenIDs are alphanumeric, and their pattern can be specified by setting the ``new_token_pattern`` property.
+
+The pattern should be a string comprised of alphanumeric characters, "#", and "@". When generating a new TokenID, instances of "#" will be replaced with a random digit 0-9 and instances of "@" will be replaced with a letter A-Z.
+
+For example, if the specified pattern is ``0123#@@1###``, a randomly generated TokenID might look like ``01238EJ1449``.
```
Comment (manthey) [inline docs/CUSTOMIZING.rst:137]: Do we want to ever generate lowercase letters?  This is a question to ask the users, not something to change in this PR.

File: wsi_deid/web_client/views/ItemList.js [hunk 2]
```diff
@@ -171,6 +181,110 @@ wrap(ItemListWidget, 'render', function (render) {
...
+        });
+        return result.join('');
+    };
+
+    const refileCheckedItems = () => {
+        const togetherSelect = $('.g-refile-select-togetherness');
+        const newOrExistingSelect = $('.g-refile-select-new-or-existing');
+        const fileTogether = togetherSelect.find(':selected').val() === 'together';
+        const useExistingToken = newOrExistingSelect.find(':selected').val() === 'existing';
+        const tokenPattern = this._wsi_deid_settings['new_token_pattern'];
```
Comment (manthey) [inline wsi_deid/web_client/views/ItemList.js:206]: We should set a default for the new_token_pattern.  If it isn't in the config file, the code throws and error.

**Comment by manthey** [comment]
One thing I'm not sure about is placement of the controls.  Currently, when the first item is checked, the controls push the list down.  Scrolling down will make then scroll off the top of the list view.  Maybe jamming the controls in the list header would make more sense:
image

But then a narrow window might have issues (haven't checked).

File: wsi_deid/web_client/views/ItemList.js (hunk 2 above)
Comment (naglepuff) [reply wsi_deid/web_client/views/ItemList.js]: See e81fcae

**Comment by naglepuff** [comment]
@manthey I've moved the controls into the hierarchy widget per your suggestion.

**Review by manthey — APPROVED** [review]

**Pull request merged**